### PR TITLE
AUTOPAY-26733: Add Servlet-friendly methods to http-signatures library

### PR DESCRIPTION
**Pull request by branislavbednar**

### CODE REVIEW

**Comment by branislavbednar** [comment]
Check related [PR](https://github.com/Visma-AutoPay/autopay-psd2/pull/247) in psd2 repo

**Review by tikre2910 — APPROVED** [review]

File: src/main/java/net/visma/autopay/http/signature/SignatureContext.java [hunk 1]
```diff
@@ -182,6 +189,28 @@ public Builder targetUri(String targetUri) {
...
+         * Sets target URI
+         * <p>
+         * Method offers compatibility with {@code HttpServletRequest} methods.
+         * {@code targetUri} is not the same as servlet's {@code requestUri}.
+         *
+         * @param requestUrl  request URL
+         * @param queryString query string
+         * @return This builder
+         */
+        public Builder targetUri(StringBuffer requestUrl, String queryString) {
```
Comment (robelcik) [inline src/main/java/net/visma/autopay/http/signature/SignatureContext.java:202]: Could you add example call? Here, and in README.md: Full example -> Creating signature -> Signature context.

Also, be prepared to write release notes. See https://github.com/Visma-AutoPay/http-signatures/releases

File: src/main/java/net/visma/autopay/http/signature/SignatureContext.java [hunk 2]
```diff
@@ -220,6 +249,48 @@ public Builder headers(Map<String, ?> headers) {
...
+        /**
+         * Adds HTTP headers.
+         * <p>
+         * Method offers compatibility with {@code HttpServletRequest} methods.
+         *
+         * @param headerNames   enumeration of HTTP servlet request header names
+         * @param headersGetter getter of HTTP servlet request headers enumeration
+         * @return This builder
+         */
+        public Builder headers(Enumeration<String> headerNames, Function<String, Enumeration<String>> headersGetter) {
```
Comment (robelcik) [inline src/main/java/net/visma/autopay/http/signature/SignatureContext.java:261]: Could you add example call? Here, and in README.md: Full example -> Creating signature -> Signature context

File: src/main/java/net/visma/autopay/http/signature/SignatureContext.java [hunk 3]
```diff
@@ -220,6 +249,48 @@ public Builder headers(Map<String, ?> headers) {
...
+        /**
+         * Adds HTTP headers.
+         * <p>
+         * Method offers compatibility with {@code HttpServletResponse} methods.
+         *
+         * @param headerNames   collection of HTTP servlet response header names
+         * @param headersGetter getter of HTTP servlet response headers collection
+         * @return This builder
+         */
+        public Builder headers(Collection<String> headerNames, Function<String, Collection<String>> headersGetter) {
```
Comment (robelcik) [inline src/main/java/net/visma/autopay/http/signature/SignatureContext.java:282]: Could you add example call?

File: src/main/java/net/visma/autopay/http/signature/SignatureContext.java (hunk 1 above)
Comment (robelcik) [inline src/main/java/net/visma/autopay/http/signature/SignatureContext.java:202]: Sonar complains about using `StringBuffer`: https://sonar.visma.com/project/issues?resolved=false&inNewCodePeriod=true&types=CODE_SMELL&pullRequest=66&id=AutoPay.HttpSignatures&open=AYzZ-Y3eSVBfSUagLExq

I think we can suppress it (in the code).

**Review by robelcik — CHANGES_REQUESTED** [review]

File: src/main/java/net/visma/autopay/http/signature/SignatureContext.java (hunk 1 above)
Comment (branislavbednar) [reply src/main/java/net/visma/autopay/http/signature/SignatureContext.java]: example call added, release notes prepared

**Pull request merged**